src/Features/CSharp/Portable/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs

```diff
@@ -74,56 +75,65 @@ public override async Task ProvideCompletionsAsync(CompletionContext context)
 
                 var span = new TextSpan(position, 0);
                 var semanticModel = await document.GetSemanticModelForSpanAsync(span, cancellationToken).ConfigureAwait(false);
-                var type = typeInferenceService.InferType(semanticModel, position,
-                    objectAsDefault: true,
+                var types = typeInferenceService.InferTypes(semanticModel, position,
                     cancellationToken: cancellationToken);
 
-                // If we have a Nullable<T>, unwrap it.
-                if (type.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T)
+                if (types.Length == 0)
                 {
-                    type = type.GetTypeArguments().FirstOrDefault();
-
-                    if (type == null)
-                    {
-                        return;
-                    }
+                    types = ImmutableArray.Create<ITypeSymbol>(semanticModel.Compilation.ObjectType);
                 }
 
-                if (type.TypeKind != TypeKind.Enum)
+                foreach (var typeIterator in types)
```

Comment:
nit: `foreach (var type in types)` is clearer since you're not reusing the iterator variable at another point.

Reply:
The only reason for this is there are assignments to the type variables in the body of the loop: 
var type = typeIterator;
 type = type.GetTypeArguments().FirstOrDefault();

---
In reply to: [292626186](https://github.com/dotnet/roslyn/pull/36293#discussion_r292626186) [](ancestors = 292626186)